ext/reflection/php_reflection.c

```diff
@@ -3778,7 +3781,42 @@ ZEND_METHOD(reflection_class, getStaticProperties)
 	}
 
 	array_init(return_value);
-	add_class_vars(ce, 1, return_value);
+
+	ZEND_HASH_FOREACH_STR_KEY_PTR(&ce->properties_info, key, prop_info) {
+		if (((prop_info->flags & ZEND_ACC_PROTECTED) &&
+		     !zend_check_protected(prop_info->ce, ce)) ||
+		    ((prop_info->flags & ZEND_ACC_PRIVATE) &&
+		     prop_info->ce != ce)) {
+			continue;
+		}
+		prop = NULL;
+		if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
+			zval *members;
+			if ((members = CE_STATIC_MEMBERS(ce))) {
```

Comment:
We can `zend_update_class_constants()` above, so this check shouldn't be necessary, and the handling for CONSTANT_AST below shouldn't be necessary either.

Reply:
Hmm, `zend_update_class_constants()` is already called on line 3777, and when I replace the else's body with `ZEND_ASSERT(0)`, I have 4 tests failing when OPcache is enabled.

Reply:
After update_class_constants, add:

```
if (!CE_STATIC_MEMBERS(ce)) {
    zend_class_init_statics(ce);
}
```

and then the rest of my comment should hold (and the provided test case be fixed).

Reply:
Thanks! Very helpful. :)